backend/tests/unit/modules/extract/test_extract.py

```diff
@@ -0,0 +1,159 @@
+import pytest
+
+from modules.extract.domain import commands, value_objects
+from modules.extract.domain.exceptions import (
+    DataValidationError,
+    FileExtensionNotSupportedError,
+)
+from modules.extract.domain.ports import repositories, units_of_work
+from modules.extract.services.commands import extract
+from tests import test_const
+
+from ....common.annotations import YieldFixture
+from .fakers import (
+    TestExtractDomainRepository,
+    TestExtractUnitOfWork,
+    TestFileDomainRepository,
+)
+
+
+def test_extract_successfully_read_csv_file(
+    test_data_unit_of_work: units_of_work.AbstractExtractUnitOfWork,
+    test_extract_repository: repositories.AbstractExtractDomainRepository,
+    test_file_repository: repositories.AbstractFileDomainRepository,
+):
+    # Given
+    test_dataset_size = 10
+
+    # When
+    input_data: value_objects.InputData = extract(
+        extract_unit_of_work=test_data_unit_of_work,
+        command=commands.ExtractData(
+            file=test_const.CORRECT_INPUT_CSV.read_bytes(),
+            file_name=test_const.CORRECT_INPUT_CSV.name,
+        ),
+    )
+
+    # Then
+    assert not input_data.empty  # type: ignore[attr-defined]
+    assert input_data.shape[0] == test_dataset_size  # type: ignore[attr-defined]
```

Comment:
What does it check?
Data structure?

Reply:
it checks if dataframe has expected 10 rows.
I changed that to len(input_data)